tests/unit/utils/test_zarr_methods.py

```diff
@@ -1337,3 +1337,347 @@ def test_update_zarr(manager_class):
         dm.update_parse_operations.assert_called_once_with(
             original_dataset, update_dataset, insert_times, append_times
         )
+
+    @staticmethod
+    def test_update_setup(manager_class, fake_original_dataset, fake_complex_update_dataset):
+        dm = manager_class()
+        insert_times, update_times = dm.update_setup(fake_original_dataset, fake_complex_update_dataset)
+        assert insert_times == [
```

Comment:
For ease of maintenance going forward could we represent the blocks of continuous dates as ranges, e.g. `numpy.arange(date, date1, timedelta64(1, “[D]”)` as you do below? And then merge them all? That’s much easier to read and understand at a glance then this big block of dates (which I built back in the day, doh).

Reply:
Since the point is to push the update mechanism to elegantly divide up contiguous ranges of dates it’s really important to quickly see those date blocks and outlier single dates.

Reply:
I almost did that, but wasn't sure it would actually be more readable given the number of non-contiguous regions. It's still kinda ugly but maybe a little better?

7f81e87

Reply:
Yeah I think that’s better. There’s no world where this is super readable but at least you can identify the breaks easily.